gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/monitor/ClouderaManagerClusterConfigurationMonitor.java

```diff
@@ -0,0 +1,876 @@
+
+  private static final ClouderaManagerServiceDiscoveryMessages log =
+      MessagesFactory.get(ClouderaManagerServiceDiscoveryMessages.class);
+
+
+  // The format of the filter employed when restart events are queried from ClouderaManager
+  private static final String RESTART_EVENTS_QUERY_FORMAT = "category==" + ApiEventCategory.AUDIT_EVENT.getValue() +
+      ";attributes.command==Restart" +
+      ";attributes.command_status==SUCCEEDED" +
+      ";attributes.cluster==\"%s\"%s";
+
+  // The format of the timestamp element of the restart events query filter
+  private static final String EVENTS_QUERY_TIMESTAMP_FORMAT = ";timeOccurred=gt=%s";
+
+  // The default amount of time before "now" the monitor will check for restart events the first time
+  private static final long DEFAULT_EVENT_QUERY_DEFAULT_TIMESTAMP_OFFSET = (60 * 60 * 1000); // one hour
+
+  // ISO 8601 datetime format for restart event query filtering
+  private static final ThreadLocal<DateFormat> eventQueryTimestampFormat =
+      ThreadLocal.withInitial(() -> new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.getDefault()));
+
+  private GatewayConfig gatewayConfig;
+  private AliasService aliasService;
+  private PollingConfigAnalyzer internalMonitor;
+  private List<ConfigurationChangeListener> changeListeners = new ArrayList<>();
+
+  // Cache of ClouderaManager API clients, keyed by discovery address
+  private final Map<String, DiscoveryApiClient> clients = new ConcurrentHashMap<>();
+
+  // ClouderaManager address
+  //    clusterName -> ServiceDiscoveryConfig
+  //
+  private final Map<String, Map<String, ServiceDiscoveryConfig>> clusterMonitorConfigurations = new ConcurrentHashMap<>();
+
+  // ClouderaManager address
+  //    clusterName
+  //        serviceType -> Properties
+  //
+  private final Map<String, Map<String, Map<String, ServiceConfigurationModel>>> clusterServiceConfigurations =
+      new ConcurrentHashMap<>();
+
+  private final ReadWriteLock serviceConfigurationsLock = new ReentrantReadWriteLock();
+
+  private final ReadWriteLock clusterMonitorConfigurationsLock = new ReentrantReadWriteLock();
+
+  // Timestamp records of the most recent restart event query per discovery address
+  private Map<String, String> eventQueryTimestamps = new ConcurrentHashMap<>();
+
+  // The amount of time before "now" the monitor will check for restart events the first time
+  private long eventQueryDefaultTimestampOffset = DEFAULT_EVENT_QUERY_DEFAULT_TIMESTAMP_OFFSET;
+
+  private ExecutorService executorService;
+
+  public static String getType() {
+    return TYPE;
+  }
+
+  ClouderaManagerClusterConfigurationMonitor(final GatewayConfig config, final AliasService aliasService) {
+    this.gatewayConfig = config;
+    this.aliasService = aliasService;
+
+    eventQueryTimestampFormat.get().setTimeZone(TimeZone.getTimeZone("UTC"));
+
+    ThreadFactory tf = (new ThreadFactoryBuilder()).setNameFormat("ClouderaManagerConfigurationMonitor-%d").build();
```

Comment:
Looks like this is the only usage of Guava. We already have a pattern for doing this with commons-lang3 instead?

https://github.com/apache/knox/blob/master/gateway-cm-integration/src/main/java/org/apache/knox/gateway/cm/descriptor/ClouderaManagerDescriptorMonitor.java#L59

Another option is potentially a custom thread factory.

Reply:
+1